Prog/Voices_q15_recode.sas

```diff
@@ -55,73 +55,326 @@ data Q15_recode_unq;
-  
-  ** Reporting areas **;
-  
-  select ( dov_urban );
-    when ( 5, 6, 7, 8 ) geo = 5;
-    otherwise geo = dov_urban;
-  end;
-    
-  race =.;
-  IF (PPETHM=1) THEN race = 1;
-  IF (PPETHM=2) THEN race = 2;
-  IF (PPETHM=4) THEN race = 3;
-  if race=. and PPETHM=3 and PPRACEM in(4, 5) then race = 4; *keeping Asian/PI only if also Non-Hispanic Other;
-  if race=. and PPETHM in (3, 5) then race=5; *all other non-Hispanic other and non-Hispanic 2+ race.; 
-   
-  format race race.;
-
-  educ =.;
-  IF (PPEDUCAT=1) or(PPEDUCAT=2) or (PPEDUCAT=3) THEN educ = 0;
-  IF (PPEDUCAT=4) THEN educ =1;
-  format educ educ.;
-
-  income =.;
-  IF (1 <= PPINCIMP <= 11) THEN income = 1;
-  IF (12 <= PPINCIMP <= 15) THEN income = 2;
-  IF (16 <= PPINCIMP <= 21) THEN income = 3;
-  format income income.;
-
-  age =.;
-  IF (18 <= PPAGE <= 34) THEN age = 1;
-  IF (35 <= PPAGE <= 64) THEN age = 2;
-  IF (PPAGE >= 65) THEN age = 3;
-  format age age.;
-
-  gender =.;
-  IF (PPGENDER=1) THEN gender = 0;
-  IF (PPGENDER=2) THEN gender = 1;
-  format gender gender.;
-
-  homeown =.;
-  IF (PPRENT=2) or (PPRENT=3)  THEN homeown = 0;
-  IF (PPRENT=1) THEN homeown = 1;
-  format homeown homeown.;
-
-  label
-    race = "Respondent's Race/Ethnicity"
-    educ = "Respondent's educational attainment"
-    income = "Respondent's HH income"
-	age = "Respondent's age"
-    gender = "Respondent's gender"
-    homeown = "Respondent's homeownership status";
-
   rename recode=Q15_recode;
   
 run;
 
+proc transpose data=Voices_Q15_notyetrecoded out=Voices_Q15_notyetrecoded_1;
+
+	by caseid;
+	var Q15_Text1 Q15_Text2 Q15_Text3; 
+
+run;
+data Voices_Q15_notyetrecoded_2 (drop=_name_);
+
+	set Voices_Q15_notyetrecoded_1 (drop=_label_);
+
+	respnum=.;
+	if _name_="Q15_Text1" then respnum=1;
+	if _name_="Q15_Text2" then respnum=2;
+	if _name_="Q15_Text3" then respnum=3;
+
+run; 
+data Voices_Q15_toreview;
+
+	set Voices_Q15_r1 (rename=(Q15_Text1=Q15_Text))
+		Voices_Q15_r2 (rename=(Q15_Text2=Q15_Text))
+		Voices_Q15_r3 (rename=(Q15_Text3=Q15_Text))
+		Voices_Q15_notyetrecoded_2(rename=(col1=Q15_Text));
+
+		Q15_text=propcase(Q15_text);
+
+		drop fuzzratio;
+run;
+%let list='ENTERTAINMENT' 'FOOD' 'CULTURE' 'MUSEUMS' 'MONUMENTS' 'PERFORM' 'ARTS' 'SPORTS' 'NIGHTLIFE' 'JOBS' 'PEOPLE' 'LOCATION' 'DIVERSITY' 'TRANSPORTATION' 'BUS' 'METRO'
+    'AIRPORTS' 'NATURE' 'COMMUNITY' 'SCHOOLS' 'WEATHER' 'LIFE' 'FAMILY' 'SHOPPING' 'SERVICES' 'CAPITAL' 'ECONOMY' 'NOTHING' 'SAFETY' 'HOUSING' 'DC' 'POLITICS' 'WALKABILITY'
+    'COSTS' '';
+
+%macro printlist;
+
+%do i=1 %to 35;
+%let var=%scan(&list,&i.," ");
+
+proc freq data=Voices_Q15_toreview;
+tables Q15_Text ;
+where  Q15_recode=&var.;
+Title "Coded as &var"; 
+
+run;
+%end; 
+
+%mend;
+
+%printlist;
+
+data Voices_Q15_newrecode (drop=diversity culture politics capital monuments transpo metro jobs food nightlife entertain services people);
+
+	set Voices_Q15_toreview;
+
+			culture=index(Q15_text,"Cultural Activ");
```

Comment:
These comparisons are all going to be case sensitive. You might be missing some items because of that.

Reply:
@ptatian - i did propcase on Q15_text before this step. That should work right?